### PR TITLE
xds: create clusters per HttpRoute (API)

### PR DESCRIPTION
**Pull request by NomadXD**
Deduplicate cluster creation by creating a cluster per HttpRoute (API). With the current implementation we use the `backendRefs` list in the `spec.rules` and all the backends listed are added as load balance endpoints in the cluster. 

Fixes: #551 

Signed-off-by: Lahiru De Silva [lahirude@wso2.com](mailto:lahirude@wso2.com)

### CODE REVIEW

File: internal/ir/xds.go [hunk 1]
```diff
@@ -176,7 +177,7 @@ type HTTPRoute struct {
 	// Redirections to be returned for this route. Takes precedence over Destinations.
 	Redirect *Redirect
 	// Destinations associated with this matched route.
-	Destinations []*RouteDestination
+	DestinationCluster *DestinationCluster
```
Comment (arkodg) [inline internal/ir/xds.go:180]: would it be easier to guarantee reuse as well as uniqueness if we added a 
`Destination string` here and created a top level `Destinations map[string][]*RouteDestination` within `Xds` ?

**Comment by danehans** [comment]
@NomadXD can you get CI to pass and fix the merge conflicts?

**Comment by danehans** [comment]
@NomadXD can you get CI to pass and fix the merge conflicts?

File: internal/ir/xds.go (hunk 1 above)
Comment (NomadXD) [reply internal/ir/xds.go:180]: yeah that's a better approach, I will update the PR.

**Pull request closed without merging**